File: apollo-converters/pojo/src/test/java/com/apollographql/android/converter/github/type/CustomType.java
package com.apollographql.android.converter.github.type;

import com.apollographql.android.api.graphql.ScalarType;

import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  DATETIME {
    public String typeName() {
      return "DateTime";
    }
  },

  URI {
    public String typeName() {
      return "URI";
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/hero_details_semantic_naming/type/CustomType.kt
package com.example.hero_details_semantic_naming.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = String::class.java
    }
}
